src/main/scala/mastermind/Mastermind.scala

```diff
@@ -13,19 +13,48 @@ val validColors = List('B', 'Y', 'R', 'G')
 
 /** Get a random color from the list of valid colors */
 def getRandomColor(): Color =
-  ???
+  val rand = new scala.util.Random
+  val ind :Int = rand.nextInt(validColors.length)
+  validColors.apply(ind)
```

Comment:
I like how each step is split here, it makes it really easy to trace what is happening in each line and I know this approach is much easier to debug.